wordpress-mobile/WordPress-Android
# Refactor notification detail activity to use ViewPager2

### Description

This PR updates the `NotificationsDetailActivity` to migrate from `ViewPager` to `ViewPager2`, to remove the usages of deprecated framework APIs. It also, in turn, migrates `NotificationDetailFragmentAdapter` to extend `FragmentStateAdapter` instead of the deprecated `FragmentStatePagerAdapter`.

#### Note

There are work-arounds in place in the existing implementation to mitigate framework bugs, but these work-arounds are no longer possible with the migration to `ViewPager2`, and likely no longer necessary.

* An `IllegalStateException` is [wrapped in a try / catch](https://github.com/wordpress-mobile/WordPress-Android/pull/20467/files#diff-e48edd5a3728c682d33322513d5ae0d968c321d8cda405d1dd45e9f4ac8b4152L632-L638) in `restoreState`, which cannot be overriden as it declared final in `FragmentStateAdapter`.
  * [Original issue](https://issuetracker.google.com/issues/36963722)
* A `TransactionTooLargeException` is avoided by [nulling the nested state collection in the bundle](https://github.com/wordpress-mobile/WordPress-Android/pull/20467/files#diff-e48edd5a3728c682d33322513d5ae0d968c321d8cda405d1dd45e9f4ac8b4152L650) in `saveState`, which is no longer possible since it is declared final in `FragmentStateAdapter`.
  * [Original issue](https://issuetracker.google.com/issues/37103380#comment77)

I don't expect these issues to reoccur with the migration to `ViewPager2`, but I'm noting these removals here, in case unexpected regressions are detected.

-----

## To Test:

Smoke test (behavior should not change):

* Log in to an account with notifications
* Tap the notifications tab
* Tap one of the notifications (expect to see the detail view)
* Swipe left and right to view other notifications in the list
* Expect the transitions to be working as before


-----

## Regression Notes

1. Potential unintended areas of impact

Notifications Detail View

2. What I did to test those areas of impact (or what existing automated tests I relied on)

Manual testing

3. What automated tests I added (or what prevented me from doing so)

The code is not yet in a state to facilitate adding unit tests for these changes.

-----

## PR Submission Checklist:


-----

## Testing Checklist (strike-out the not-applying and unnecessary ones):
